frontend/src/components/AddOrEditStep.vue

```diff
@@ -1,75 +1,104 @@
 <script setup lang="ts">
+import { computed, watch } from "vue";
+import { TRACK_FIELD_INPUT, TRACK_FIELD_TYPE } from "@/constants";
+
 import { useTrackStore } from "@/store/track";
 import { useStepStore } from "@/store/step";
-import { TRACK_FIELD_INPUT, TRACK_FIELD_TYPE } from "@/constants";
-import { ref } from "vue";
 
 const trackStore = useTrackStore();
 const stepStore = useStepStore();
 
-let showAddStepModal = ref(false);
+const props = defineProps({
+  show: {
+    type: Boolean,
+    default: false,
+  },
+  edit: {
+    type: Boolean,
+    default: false,
+  },
+});
+
+const emit = defineEmits(["close"]);
+
+// NOTE: this is a little weird, but props.show is _not_ reactive, so we cannot watch it
+//   directly and wrapping it in toRef would be misleading, since its readonly, so
+//   computed is a fair compromise
+const show = computed(() => props.show);
+
+watch(show, (newValue) => {
+  if (newValue) {
+    if (!props.edit) {
```

Comment:
Would it make sense to merge this to `if`s?

Reply:
Yeah totally, this is just a leftover. The inner if used to be in it's own function. I will move them together.